File: InterviewBit/Strings/LongestCommonPrefix.js
const commonChars = (s1, s2) => {
  let res = "";
  let n = Math.min(s1.length, s2.length);
  for (let i = 0; i < n; i++) {
    if (s1[i] === s2[i]) {
      res += s1[i];
    } else {
      break;
    }
  }
  return res;
};
const longestCommonPrefix = (a) => {
  let res = a[0];
  for (let i = 1; i < a.length; i++) {
    res = commonChars(res, a[i]);
  }
  return res;
};
let a = ["abcdefgh", "abefghijk", "abcefgh"];
console.log(longestCommonPrefix(a));

File: InterviewBit/Strings/LongestCommonPrefix.java
package InterviewBit.Strings;

import java.util.Arrays;

public class LongestCommonPrefix {
    public static String longestCommonPrefix(String[] s) {
        StringBuilder ans = new StringBuilder();
        if (s.length == 0) {
            return "";
        }
        Arrays.sort(s);
        char[] first = s[0].toCharArray();
        char[] last = s[s.length - 1].toCharArray();
        for (int i = 0; i < first.length; i++) {
            if (first[i] != last[i]) {
                break;
            }
            ans.append(first[i]);
        }
        return ans.toString();
    }

    public static void main(String[] args) {
        String[] s = { "abcdefgh", "aefghijk", "abcefgh" };
        System.out.println(longestCommonPrefix(s));
    }
}
